app/controllers/vendors_controller.rb

```diff
@@ -0,0 +1,33 @@
+class VendorsController < ApplicationController
+  before_filter :authenticate_user!
+  load_and_authorize_resource
+
+  def index
+    @vendors = Vendor.all
+  end
+
+  def show
+    @vendor = Vendor.find(params[:id])
```

Comment:
this may result in double loading since `load_and_authorize_resource` loads and authorizes `Vendor.find(params[:id])` and assigns it to `@vendor`. I think you can remove it and the tests will continue to pass?

Reply:
yep the tests still pass